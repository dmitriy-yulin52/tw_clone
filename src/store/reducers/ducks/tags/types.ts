
export enum LoadingState{
    LOADED = 'LOADED',
    LOADING = 'LOADING',
    ERROR = 'ERROR',
    NEVER = 'NEVER',

}

export interface Tag {
    id:string,
    fullName:string
    userName:string
    count:number
    text:string
}

export interface TagsState {
    items:Tag[]
    loadingState:LoadingState

}