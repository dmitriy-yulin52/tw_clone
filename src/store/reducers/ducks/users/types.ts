
export enum LoadingState{
    LOADED = 'LOADED',
    LOADING = 'LOADING',
    ERROR = 'ERROR',
    NEVER = 'NEVER',

}

export interface User {
    id:string,
    name:string
    userName:string
    avatarUrl:string
}

export interface UsersState {
    items:User[]
    loadingState:LoadingState

}