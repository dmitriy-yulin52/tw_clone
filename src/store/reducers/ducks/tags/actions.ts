import {Action} from 'redux'
import {LoadingState, Tag} from "./types";


export enum TagsActionsType {
    SET_ITEMS = 'tags/SET_ITEMS',
    FETCH_ITEMS = 'tags/FETCH_ITEMS',
    SET_LOADING_STATE = 'tags/SET_LOADING_STATE'
}

export type TagsActions = SetTagsType | SetTagsLoadingStateType | FetchTagsType

/** types-actions*/
export interface SetTagsType extends Action<TagsActionsType> {
    type: TagsActionsType.SET_ITEMS
    payload: Tag[]
}
export interface FetchTagsType extends Action<TagsActionsType> {
    type: TagsActionsType.FETCH_ITEMS
}
export interface SetTagsLoadingStateType extends Action<TagsActionsType> {
    type: TagsActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setTweets = (payload: Tag[]): SetTagsType => {

    console.log(payload,'payload')
    return {
        type: TagsActionsType.SET_ITEMS,
        payload
    }
}
export const setTweetsLoadingState = (payload:LoadingState): SetTagsLoadingStateType => {
    return {
        type: TagsActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchTweets = (): FetchTagsType => {
    return {
        type: TagsActionsType.FETCH_ITEMS,
    }
}
