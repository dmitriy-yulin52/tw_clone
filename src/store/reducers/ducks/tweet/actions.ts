import {Action} from 'redux'
import {LoadingState} from "./types";
import {Tweet} from "../tweets/types";


export enum TweetDataActionsType {
    SET_TWEET_DATA = 'tweet/SET_DATA',
    FETCH_TWEET_DATA = 'tweet/FETCH_DATA',
    SET_LOADING_STATE = 'tweet/SET_LOADING_STATE'
}

export type TweetDataActions = SetTweetDataType | SetTweetDataLoadingStateType | FetchTweetDataType

/** types-actions*/
export interface SetTweetDataType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.SET_TWEET_DATA
    payload: Tweet
}
export interface FetchTweetDataType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.FETCH_TWEET_DATA
    payload:string
}
export interface SetTweetDataLoadingStateType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setTweetData = (payload: Tweet): SetTweetDataType => {
    console.log(payload,'setTweetData')
    return {
        type: TweetDataActionsType.SET_TWEET_DATA,
        payload
    }
}
export const setTweetDataLoadingState = (payload:LoadingState): SetTweetDataLoadingStateType => {
    return {
        type: TweetDataActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchTweetData = (payload:string): FetchTweetDataType => {
    return {
        type: TweetDataActionsType.FETCH_TWEET_DATA,
        payload
    }
}
