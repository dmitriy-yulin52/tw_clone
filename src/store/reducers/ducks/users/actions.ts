import {Action} from 'redux'
import {LoadingState, User} from "./types";


export enum UsersActionsType {
    SET_ITEMS = 'users/SET_ITEMS',
    FETCH_ITEMS = 'users/FETCH_ITEMS',
    SET_LOADING_STATE = 'users/SET_LOADING_STATE'
}

export type UsersActions = SetUsersType | SetUsersLoadingStateType | FetchUsersType

/** types-actions*/
export interface SetUsersType extends Action<UsersActionsType> {
    type: UsersActionsType.SET_ITEMS
    payload: User[]
}
export interface FetchUsersType extends Action<UsersActionsType> {
    type: UsersActionsType.FETCH_ITEMS
}
export interface SetUsersLoadingStateType extends Action<UsersActionsType> {
    type: UsersActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setUsers = (payload: User[]): SetUsersType => {

    console.log(payload,'payload')
    return {
        type: UsersActionsType.SET_ITEMS,
        payload
    }
}
export const setUsersLoadingState = (payload:LoadingState): SetUsersLoadingStateType => {
    return {
        type: UsersActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchUsers = (): FetchUsersType => {
    return {
        type: UsersActionsType.FETCH_ITEMS,
    }
}
