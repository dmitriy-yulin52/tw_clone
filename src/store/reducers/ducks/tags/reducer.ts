import {LoadingState, TagsState} from "./types";
import produce, {Draft} from "immer";
import {TagsActions, TagsActionsType} from "./actions";


const initialTweetsState: TagsState = {
    items: [],
    loadingState: LoadingState.NEVER
}


export const tagsReducer = produce((draft: Draft<TagsState>, action: TagsActions) => {


    switch (action.type) {
        case TagsActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case TagsActionsType.SET_ITEMS:
            draft.items = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case TagsActionsType.FETCH_ITEMS:
            draft.items = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialTweetsState)