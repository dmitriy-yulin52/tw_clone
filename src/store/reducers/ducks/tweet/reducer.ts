import {LoadingState, TweetDataState} from "./types";
import produce, {Draft} from "immer";
import {TweetDataActions, TweetDataActionsType} from "./actions";
import {Tweet} from "../tweets/types";


const initialTweetDataState: TweetDataState = {
    data: null,
    loadingState: LoadingState.NEVER
}


export const tweetReducer = produce((draft: Draft<TweetDataState>, action: TweetDataActions) => {


    switch (action.type) {
        case TweetDataActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case TweetDataActionsType.SET_TWEET_DATA:
            draft.data = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case TweetDataActionsType.FETCH_TWEET_DATA:
            // draft.data = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialTweetDataState)