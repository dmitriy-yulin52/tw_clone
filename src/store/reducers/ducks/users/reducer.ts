import {LoadingState, UsersState} from "./types";
import produce, {Draft} from "immer";
import {UsersActions, UsersActionsType} from "./actions";


const initialUsersState: UsersState = {
    items: [],
    loadingState: LoadingState.NEVER
}


export const usersReducer = produce((draft: Draft<UsersState>, action: UsersActions) => {


    switch (action.type) {
        case UsersActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case UsersActionsType.SET_ITEMS:
            draft.items = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case UsersActionsType.FETCH_ITEMS:
            draft.items = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialUsersState)