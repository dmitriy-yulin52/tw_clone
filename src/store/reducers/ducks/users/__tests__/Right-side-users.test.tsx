import {setUsers} from "../actions";
import {usersReducer} from "../reducer";
import {User, UsersState} from "../types";
import {LoadingState} from "../../tweets/types";

const users_test: User[] = [
    {
        "id": "1001",
        "name": "Ivanov",
        "userName": "ivanka",
        "avatarUrl": "https://jooinn.com/images/man-standing-on-street.jpg"
    },
    {
        "id": "1002",
        "name": "Therese",
        "userName": "queen",
        "avatarUrl": "https://jooinn.com/images/man-standing-on-street.jpg"
    }
]


describe('users test', () => {

    test('usersReducer', () => {

        const initialUsersState: UsersState = {
            items: [],
            loadingState: LoadingState.NEVER
        }

        const data = {
            payload: users_test,
            type: 'users/SET_ITEMS',
        }
        const action = setUsers(users_test)
        const end_state = usersReducer(initialUsersState, action)

        expect(action).toEqual(data)
        expect(end_state.items[0].name).toBe('Ivanov')
        expect(end_state.items[1].name).toBe('Therese')
        expect(end_state.items.length).toEqual(2)
        expect(end_state).toMatchSnapshot()
    })
})