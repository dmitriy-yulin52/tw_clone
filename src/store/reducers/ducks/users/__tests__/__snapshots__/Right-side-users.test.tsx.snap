// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users test usersReducer 1`] = `
Object {
  "items": Array [
    Object {
      "avatarUrl": "https://jooinn.com/images/man-standing-on-street.jpg",
      "id": "1001",
      "name": "Ivanov",
      "userName": "ivanka",
    },
    Object {
      "avatarUrl": "https://jooinn.com/images/man-standing-on-street.jpg",
      "id": "1002",
      "name": "Therese",
      "userName": "queen",
    },
  ],
  "loadingState": "LOADED",
}
`;
